content/realtime/messages.textile

```diff
@@ -420,7 +420,13 @@ h6(#data).
   default: data
   csharp: Data
 
-The message payload, if provided.<br>__Type: <span lang="default">@String@, @StringBuffer@, @JSON Object@</span><span lang="java">@String@, @ByteArray@, @JSONObject@, @JSONArray@</span><span lang="csharp">@String@, @byte[]@, @plain C# object that can be serialized to Json@</span><span lang="ruby">@String@, @Binary@ (ASCII-8BIT String), @Hash@, @Array@</span><span lang="python">@String@, @Bytearray@, @Dict@, @List@</span><span lang="php">@String@, @Binary String@, @Associative Array@, @Array@</span><span lang="objc">@NSString *@, @NSData *@, @NSDictionary *@, @NSArray *@</span><span lang="swift">@String@, @NSData@, @Dictionary@, @Array@</span>__
+The message payload, if provided.<br>__Type: <span lang="default">@String@, @StringBuffer@, @JSON Object@</span><span lang="java">@String@, @ByteArray@, @JSONObject@, @JSONArray@</span><span lang="csharp">@String@, @byte[]@, @plain C# object that can be serialized to JSON@</span><span lang="ruby">@String@, @Binary@ (ASCII-8BIT String), @Hash@, @Array@</span><span lang="python">@String@, @Bytearray@, @Dict@, @List@</span><span lang="php">@String@, @Binary String@, @Associative Array@, @Array@</span><span lang="objc">@NSString *@, @NSData *@, @NSDictionary *@, @NSArray *@</span><span lang="swift">@String@, @NSData@, @Dictionary@, @Array@</span>__
+
+h6(#extras).
+  default: extras
+  csharp: Extras
+
+Metadata and/or ancillary payloads, if provided.<br>__Type: <span lang="java">@JSONObject@, @JSONArray@</span><span lang="csharp">plain C# object that can be converted to JSON</span><span lang="jsall">@JSON Object@</span><span lang="ruby">@Hash@, @Array@</span><span lang="python">@Dict@, @List@</span><span lang="swift">@Dictionary@, @Array@</span><span lang="objc">@NSDictionary *@, @NSArray *@</span><span lang="php">@Associative Array@, @Array@</span>__
```

Comment:
@SimonWoolf @paddybyers aren't there restrictions in terms of what is allowed in `extras`?  Also, @tomczoink I think we should mention that extras is used for push notifications and link to that.

Reply:
Yeah, this kinda makes it sound like it's just for any random arbitrary metadata that the user might want to send. We should make it clear that it's all reserved and should only be used for specific purposes and with specific fields that we define, of which currently the only one is `push`.

(This is the danger of taking the client lib spec and trying to expand it into user documentation -- from a client lib developer's POV, `extras` is arbitrary metadata, because client libs shouldn't be touching or validating it; but that doesn't mean anything's allowed from the user's POV)